pkg/kubelet/dockertools/docker_network.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertools
+
+import (
+	"fmt"
+	"sync"
+
+	dockertypes "github.com/docker/engine-api/types"
+	"github.com/golang/glog"
+
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/network"
+)
+
+type dockerNetwork struct {
```

Comment:
This is essentially a network plugin wrapper which helps to synchronize the calls. I feels like it can be more generic than dockerNetwork, maybe NetworkPluginManager? The only docker specific stuff here is the docker network mode. How about this? only wrap GetPodNetworkStatus here and leave determineContainerIP in dockermanager?